File: Semana02/Lab02Prueba/Matricula.java
import java.util.Scanner;

public class Matricula {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        // ================== COMMIT 1: DATOS DE ENTRADA ==================
        System.out.print("Ingrese su nombre completo: ");
        String nombreEstudiante = sc.nextLine();

        System.out.print("Ingrese el valor de cada credito (S/): ");
        double valorCredito = sc.nextDouble();
        sc.nextLine();

        System.out.print("Ingrese la cantidad de cursos a matricular: ");
        int cantidadCursos = sc.nextInt();
        sc.nextLine();

        String[] nombreCursos = new String[cantidadCursos];
        int[] creditosCursos = new int[cantidadCursos];

        for (int i = 0; i < cantidadCursos; i++) {
            System.out.print("Nombre del curso " + (i + 1) + ": ");
            nombreCursos[i] = sc.nextLine();

            System.out.print("Creditos del curso " + (i + 1) + ": ");
            creditosCursos[i] = sc.nextInt();
            sc.nextLine();
        }
        // ================== COMMIT 2: CALCULOS ==================
        double[] costoCursos = new double[cantidadCursos];
        int totalCreditos = 0;
        double totalPagar = 0;

        for (int i = 0; i < cantidadCursos; i++) {
            costoCursos[i] = creditosCursos[i] * valorCredito;
            totalCreditos += creditosCursos[i];
            totalPagar += costoCursos[i];
        }

        String cargaAcademica;
        if (totalCreditos <= 12) {
            cargaAcademica = "Malla Regular";
        } else if (totalCreditos <= 18) {
            cargaAcademica = "Carga Completa";
        } else {
            cargaAcademica = "Permiso Autorizado";
        }

        int numeroCuotas;
        if (totalPagar > 2500) {
            numeroCuotas = 3;
        } else {
            numeroCuotas = 2;
        }
        double montoCuota = totalPagar / numeroCuotas;
        // ================== COMMIT 3: RESULTADO FINAL ==================
        System.out.println();
        System.out.println("Estudiante: " + nombreEstudiante);
        System.out.println();

        System.out.println("+---------------------------+------------+------------+");
        System.out.printf("| %-27s | %-10s | %-10s |%n", "Curso", "Creditos", "Costo");
        System.out.println("+---------------------------+------------+------------+");
        for (int i = 0; i < cantidadCursos; i++) {
            System.out.printf("| %-27s | %-10d | %-10.0f |%n", nombreCursos[i], creditosCursos[i], costoCursos[i]);
        }
        System.out.println("+---------------------------+------------+------------+");

        System.out.println();
        System.out.println("Cursos Matriculados: " + cantidadCursos);
        System.out.println("Total Creditos: " + totalCreditos);
        System.out.println("Total a pagar: " + (int) totalPagar);
        System.out.println("Carga academica: " + cargaAcademica);
        System.out.println("Forma de pago: " + numeroCuotas + " cuotas de " + (int) montoCuota + " Soles");
        sc.close();
    }
}

File: Semana02/Lab02Prueba/Matricula.kt
fun main() {

    print("Ingrese su nombre completo: ")
    val nombreEstudiante = readLine() ?: ""

    print("Ingrese su turno (M=Manana, T=Tarde, N=Noche): ")
    val turno = readLine()!!.uppercase()

    print("Ingrese el valor de cada credito (S/): ")
    val valorCredito = readLine()!!.toDouble()

    print("Ingrese la cantidad de cursos a matricular: ")
    val cantidadCursos = readLine()!!.toInt()

    var totalCreditos = 0
    var totalPagar = 0.0
    var tablaCursos = ""

    for (i in 1..cantidadCursos) {

        print("Nombre del curso $i: ")
        val nombreCurso = readLine() ?: ""

        print("Creditos del curso $i: ")
        val creditosCurso = readLine()!!.toInt()

        val costoCurso = creditosCurso * valorCredito

        totalCreditos += creditosCurso
        totalPagar += costoCurso

        tablaCursos += String.format(
            "| %-27s | %-10d | %-10.0f |\n",
            nombreCurso,
            creditosCurso,
            costoCurso
        )
    }

    val cargaAcademica: String

    if (totalCreditos <= 12) {
        cargaAcademica = "Malla Regular"
    } else if (totalCreditos <= 18) {
        cargaAcademica = "Carga Completa"
    } else {
        cargaAcademica = "Permiso Autorizado"
    }

    val numeroCuotas: Int

    if (totalPagar > 2500) {
        numeroCuotas = 3
    } else {
        numeroCuotas = 2
    }

    val montoCuota = totalPagar / numeroCuotas

    var porcentajeTurno = 0.0

    if (turno == "M") {
        porcentajeTurno = 0.10
    } else if (turno == "T") {
        porcentajeTurno = 0.15
    } else if (turno == "N") {
        porcentajeTurno = 0.20
    }

    val montoTurno = totalPagar * porcentajeTurno

    val totalConTurno = totalPagar + montoTurno

    println()
    println("Estudiante: $nombreEstudiante")
    println()

    println("+-----------------------------+------------+------------+")
    println(
        String.format(
            "| %-27s | %-10s | %-10s |",
            "Curso",
            "Creditos",
            "Costo"
        )
    )
    println("+-----------------------------+------------+------------+")

    print(tablaCursos)

    println("+-----------------------------+------------+------------+")

    println()
    println("Cursos Matriculados: $cantidadCursos")
    println("Total Creditos: $totalCreditos")
    println("Total cursos: ${totalPagar.toInt()}")
    println("Turno: $turno")
    println("Recargo por turno: ${montoTurno.toInt()}")
    println("Total a pagar: ${totalConTurno.toInt()}")
    println("Carga academica: $cargaAcademica")
    println(
        "Forma de pago: $numeroCuotas cuotas de ${montoCuota.toInt()} Soles"
    )
}